src/orion/core/io/experiment_builder.py

```diff
@@ -0,0 +1,262 @@
+  argument to `orion` itself as well as the user's script name and its arguments.
+
+"""
+import logging
+
+from orion.core.io import resolve_config
+from orion.core.io.database import Database, DuplicateKeyError
+from orion.core.worker.experiment import Experiment, ExperimentView
+
+
+log = logging.getLogger(__name__)
+
+
+class ExperimentBuilder(object):
+    """Builder for :class:`orion.core.worker.experiment.Experiment`
+    and :class:`orion.core.worker.experiment.ExperimentView`
+
+    .. seealso::
+
+        `orion.core.io.experiment_builder` for more information on the process of building
+        experiments.
+
+        :class:`orion.core.worker.experiment.Experiment`
+        :class:`orion.core.worker.experiment.ExperimentView`
+    """
+
+    # pylint:disable=no-self-use
+    def fetch_default_options(self):
+        """Get dictionary of default options"""
+        return resolve_config.fetch_default_options()
+
+    # pylint:disable=no-self-use
+    def fetch_env_vars(self):
+        """Get dictionary of environment variables specific to Oríon"""
+        return resolve_config.fetch_env_vars()
+
+    def fetch_file_config(self, cmdargs):
+        """Get dictionary of options from configuration file provided in command-line"""
+        return resolve_config.fetch_config(cmdargs)
+
+    def fetch_config_from_db(self, cmdargs):
+        """Get dictionary of options from experiment found in the database
+
+        Note
+        ----
+            This method builds an experiment view in the background to fetch the configuration from
+            the database.
+        """
+        try:
+            experiment_view = self.build_view_from(cmdargs)
+        except ValueError as e:
+            if "No experiment with given name" in str(e):
+                return {}
+
+        return experiment_view.configuration
+
+    def fetch_metadata(self, cmdargs):
+        """Infer rest information about the process + versioning"""
+        return resolve_config.fetch_metadata(cmdargs)
+
+    def fetch_full_config(self, cmdargs, use_db=True):
+        """Get dictionary of the full configuration of the experiment.
+
+        .. seealso::
+
+            `orion.core.io.experiment_builder` for more information on the hierarchy of
+            configurations.
+
+        Parameters
+        ----------
+        cmdargs:
+
+        use_db: bool
+            Use experiment configuration found in database if True. Defaults to True.
+
+        Note
+        ----
+            This method builds an experiment view in the background to fetch the configuration from
+            the database.
+
+        """
+        default_options = self.fetch_default_options()
+        env_vars = self.fetch_env_vars()
+        if use_db:
+            config_from_db = self.fetch_config_from_db(cmdargs)
+        else:
+            config_from_db = {}
+        cmdconfig = self.fetch_file_config(cmdargs)
+        metadata = dict(metadata=self.fetch_metadata(cmdargs))
```

Comment:
I still maintain that we should use `{metadata=self.fetch_metadata(cmdargs)}` but this is just nitpicking ;)